### ISSUE TITLE
enabling grpclb with alts can log channel and eventLoop mismatch warning

### ISSUE THREAD

**Issue by creamsoup** [grpc/grpc-java#5986]
With grpclb and alts, name resolver creates handshaker factory which only sets the eventloop type. this can spam log about eventLoop and channel type mismatch. This warning is legit but may confuse user.

```
java.lang.Throwable
at io.grpc.netty.shaded.io.grpc.netty.NettyChannelBuilder.buildTransportFactory(NettyChannelBuilder.java:420)
at io.grpc.internal.AbstractManagedChannelImplBuilder.build(AbstractManagedChannelImplBuilder.java:507)
at io.grpc.alts.HandshakerServiceChannel$ChannelResource.create(HandshakerServiceChannel.java:62)
at io.grpc.alts.HandshakerServiceChannel$ChannelResource.create(HandshakerServiceChannel.java:46)
at io.grpc.internal.SharedResourceHolder.getInternal(SharedResourceHolder.java:104)
at io.grpc.internal.SharedResourceHolder.get(SharedResourceHolder.java:74)
at io.grpc.internal.SharedResourcePool.getObject(SharedResourcePool.java:35)
at io.grpc.alts.internal.AltsProtocolNegotiator$LazyChannel.get(AltsProtocolNegotiator.java:127)
at io.grpc.alts.ComputeEngineChannelBuilder$ProtocolNegotiatorFactory$1.newHandshaker(ComputeEngineChannelBuilder.java:104)
at io.grpc.alts.internal.AltsProtocolNegotiator$1ClientAltsProtocolNegotiator.newHandler(AltsProtocolNegotiator.java:66)
at io.grpc.alts.internal.GoogleDefaultProtocolNegotiator.newHandler(GoogleDefaultProtocolNegotiator.java:60)
at io.grpc.netty.shaded.io.grpc.netty.NettyClientTransport.start(NettyClientTransport.java:221)
at io.grpc.internal.ForwardingConnectionClientTransport.start(ForwardingConnectionClientTransport.java:33)
at io.grpc.internal.ForwardingConnectionClientTransport.start(ForwardingConnectionClientTransport.java:33)
at io.grpc.internal.InternalSubchannel.startNewTransport(InternalSubchannel.java:251)
at io.grpc.internal.InternalSubchannel.access$400(InternalSubchannel.java:65)
at io.grpc.internal.InternalSubchannel$2.run(InternalSubchannel.java:196)
at io.grpc.SynchronizationContext.drain(SynchronizationContext.java:95)
at io.grpc.SynchronizationContext.execute(SynchronizationContext.java:127)
at io.grpc.internal.ManagedChannelImpl$NameResolverListener.onResult(ManagedChannelImpl.java:1365)
at io.grpc.internal.DnsNameResolver$Resolve.resolveInternal(DnsNameResolver.java:311)
at io.grpc.internal.DnsNameResolver$Resolve.run(DnsNameResolver.java:213)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
at java.lang.Thread.run(Thread.java:748)
```

**Comment by WeiranFang** [comment]
More specifically, the HandshakerServiceChannel uses NIO type event loop for the underlying channel: https://github.com/grpc/grpc-java/blob/6e2bb6b4024356b6e4fb769bd7838df2854b28ed/alts/src/main/java/io/grpc/alts/HandshakerServiceChannel.java#L56-L57
This may need to be fixed if Epoll event loop is now a default type.

**Comment by creamsoup** [comment]
resolved by #5987
